ci/ray_ci/pipeline/gap_filling_scheduler.py

```diff
@@ -29,6 +29,41 @@ def __init__(
         self.repo_checkout = repo_checkout
         self.days_ago = days_ago
 
+    def run(self) -> List[str]:
+        """
+        Create gap filling builds for the latest failing build. If dry_run is True,
+        print the commits for each build but no builds will actually be created.
+        """
+        commits = self.get_gap_commits()
+
+        return [self._trigger_build(commit) for commit in commits]
+
+    def get_gap_commits(self) -> List[str]:
+        """
+        Return the list of commits between the latest passing and failing builds.
+        """
+        failing_revision = self._get_latest_commit_for_build_state("failed")
+        passing_revision = self._get_latest_commit_for_build_state("passed")
+        return (
+            subprocess.check_output(
+                [
+                    "git",
+                    "rev-list",
+                    "--reverse",
+                    f"^{passing_revision}",
+                    f"{failing_revision}~",
+                ],
+                cwd=self.repo_checkout,
+            )
+            .decode("utf-8")
+            .strip()
+            .split("\n")
+        )
+
+    def _trigger_build(self, commit: str) -> int:
```

Comment:
what is the return value of this? should this return an integer as a place holder so that it respects the typing annotation?

we should probably turn on typing lints for CI python scripts..